orbit/cmd/orbit/orbit.go

```diff
@@ -939,44 +936,81 @@ func (d *desktopRunner) interrupt(err error) {
+	// Hostname is the device's hostname (extracted from `system_info` osquery table).
+	Hostname string `json:"hostname"`
+	// Platform is the device's platform as defined by osquery (extracted from `os_version` osquery table).
+	Platform string `json:"platform"`
+}
+
+// getHostInfo retrieves system information about the host.
+//
+// On macOS and Linux it shells out to osqueryd to retrieve the information.
+//
+// On Windows:
+//
+//   - HardwareUUID is retrieved by shelling out to wmic, if that fails
+//     then the windows API are used.
+//   - HardwareSerial is currently not retrieved for Windows devices.
+//   - Hostname is retrieved using stdlib method.
+//   - Platform is always "windows" for windows hosts.
+//
+// NOTE: Windows uses a different approach to retrieve the device information
+// as there were issues at the time with shelling out to osquery - from what the
+// team remembers it would sometimes fail due to the osquery process not being ready yet.
+// A recent CI run without the Windows special-case did succeed, but since we don't
+// need the serial number for Windows at the moment, we opted to keep the
+// code as it is.
+func getHostInfo(osqueryPath string, osqueryDBPath string) (fleet.OrbitHostInfo, error) {
 	if runtime.GOOS == "windows" {
-		// NOTE: Windows uses a different approach as there were issues at the time
-		// with shelling out to osquery - from what the team remembers it would
-		// sometimes fail due to the osquery process not being ready yet. A recent
-		// CI run without the Windows special-case did succeed, but since we don't
-		// need the serial number for Windows at the moment, we opted to keep the
-		// code as it is.
 		uuidData, uuidSource, err := platform.GetSMBiosUUID()
 		if err != nil {
-			return "", "", err
+			return fleet.OrbitHostInfo{}, err
 		}
-
-		log.Debug().Msgf("UUID source was %s.", uuidSource)
-		return uuidData, "", nil
+		log.Debug().Str("source", string(uuidSource)).Msg("UUID")
+		// Hostname might differ from the one provided by osquery but we are sending it
+		// for troubleshooting purposes and to avoid empty host entries in the UI.
+		hostname, err := os.Hostname()
+		if err != nil {
+			return fleet.OrbitHostInfo{}, err
+		}
+		return fleet.OrbitHostInfo{
+			HardwareUUID:   uuidData,
+			HardwareSerial: "", // currently not needed for Windows.
+			Hostname:       hostname,
+			Platform:       "windows",
+		}, nil
 	}
-	type Output struct {
-		UuidString     string `json:"uuid"`
-		HardwareSerial string `json:"hardware_serial"`
+	const systemQuery = "SELECT si.uuid, si.hardware_serial, si.hostname, os.platform FROM system_info si, os_version os"
+	args := []string{
+		"-S",
+		"--database_path", osqueryDBPath,
+		"--json", systemQuery,
 	}
-
-	args := []string{"-S", "--json", "select uuid, hardware_serial from system_info"}
+	log.Debug().Str("query", systemQuery).Msg("running single query")
 	out, err := exec.Command(osqueryPath, args...).Output()
```

Comment:
Are we not using context (ctx) for calling commands in Orbit?

Reply:
I believe there's no need to in this case. Let me know if you find a reason to use it. (If this `exec.Command` fails then all of Orbit fails to run.)

Reply:
We agreed to look at this issue as a whole thing outside this PR